x-pack/plugins/fleet/common/services/agent_status.ts

```diff
@@ -61,10 +61,15 @@ export const AGENT_UPDATING_TIMEOUT_HOURS = 2;
 
 export function isStuckInUpdating(agent: Agent): boolean {
   return (
-    agent.status === 'updating' &&
-    !!agent.upgrade_started_at &&
-    !agent.upgraded_at &&
-    Date.now() - Date.parse(agent.upgrade_started_at) >
-      AGENT_UPDATING_TIMEOUT_HOURS * 60 * 60 * 1000
+    (agent.status !== 'offline' && agent.active && isAgentInFailedUpgradeState(agent)) ||
```

Comment:
after the latest discussion [here](https://github.com/elastic/kibana/issues/171419#issuecomment-1863032626) changed the logic to consider stuck in updating if upgrade failed, even though agent is back to online